File: backend/app/schemas/enrollment.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class EnrollmentCreate(BaseModel):
    course_id: int


class EnrollmentOut(BaseModel):
    id: int
    user_id: int
    course_id: int
    enrolled_at: Optional[datetime] = None
    completed: bool = False
    course_title: Optional[str] = None
    is_active: bool = True

    model_config = {"from_attributes": True}


class BulkEnrollRequest(BaseModel):
    course_id: int
    user_ids: List[int]

File: frontend-next/types/enrollment.ts
// TODO: once backend exposes EnrollmentOut in OpenAPI,
// re-export from lib/api/schema.ts instead of re-declaring here.

export type Enrollment = {
  id?: number;
  user_id?: number | null;
  user_name?: string | null;
  user_email?: string | null;
  course_id?: number;
  course_title?: string | null;
  category?: string | null;
  progress_pct?: number | null;
  completed?: boolean | null;
  total_lessons?: number | null;
  completed_lessons?: number | null;
  enrolled_at?: string | null;
};

export type AvailableCourse = {
  id: number;
  title: string;
  description?: string | null;
  category?: string | null;
  lesson_count?: number | null;
  duration_min?: number | null;
  is_active?: boolean | null;
};
